### ISSUE TITLE
[FEAT REQ] Add new check to detect items.remove into for loop from same items variable

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#1238]
### Steps to reproduce
```python
items = range(1, 6)
for item in items:
    print(item)
    items.remove(item)
```

We have the following print output:
1
3
5

A fix of this code is add `list(items)`:
```diff
items = range(1, 6)
- for item in items:
+ for item in list(items):
    print(item)
    items.remove(item)
```

Now you have the following print output:
1
2
3
4
5

### Current behavior
If we run pylint here is not emitted a message.

### Expected behavior
Feature request: Add a new check in order to detect this wrong and common cases.

Is this a good new check for pylint?

**Comment by rogalski** [comment]
For me, it seems like a generic message `mutable-sequence-modified-in-loop`.

Lists (and sets) are fairly easy. Dictionaries will be slightly trickier, since you often iterate over method call result, but you actually modify a dictionary itself.

I wonder if there's a reasonable way of detecting when modification actually occurs. I'm afraid that there would have to be a hardcoded list of "modifying" calls for each type (dict, set, list, other stdlib objects?). Some will be shared, other will be exclusive to given type.

For sure we should handle:
- method calls (pop for dict/list, remove etc.)
- del statements

Anything more I'm missing?

**Comment by PCManticore** [comment]
Sounds good to me as well.

**Comment by PCManticore** [comment]
Let's close this. I don't think we'll get to implement this one in a such a way that it will work correctly all the time on non-trivial cases.
